packages/react-native/Libraries/AppDelegate/RCTAppDelegate.mm

```diff
@@ -142,19 +144,22 @@ - (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
   return nil;
 }
 
-- (NSDictionary *)prepareInitialProps
+- (NSDictionary *)extraInitialProps
 {
   NSMutableDictionary *initProps = self.initialProps ? [self.initialProps mutableCopy] : [NSMutableDictionary new];
-
 #ifdef RCT_NEW_ARCH_ENABLED
   // Hardcoding the Concurrent Root as it it not recommended to
   // have the concurrentRoot turned off when Fabric is enabled.
   initProps[kRNConcurrentRoot] = @([self fabricEnabled]);
 #endif
-
   return initProps;
 }
```

Comment:
I'd move this to a separate static method in this file (eg `updateInitialProps`), which takes the initialProps provided, and mutates them (if necessary)